crates/rwasm/machine/src/operations/range.rs

```diff
@@ -145,4 +145,176 @@ impl<T: Copy, const START: u32, const END: u32> Range16bCols<T, START, END> {
             cols.hi_is_eq_ub_hi,
         );
     }
+
+    /// The prove always have known whether a column of rangechecker is real or not.
+    /// So the input do check should always  equals rangechecker.is_real.
+    pub fn do_range_check<AB: SP1AirBuilder>(
```

Comment:
maybe make sense to add is_real: bool to populate (not to columns), and remove populate_value